drivers/SmartThings/jbl/src/jbl/api.lua

```diff
@@ -0,0 +1,165 @@
+local log = require "log"
+local json = require "st.json"
+local RestClient = require "lunchbox.rest"
+local utils = require "st.utils"
+local cosock = require "cosock"
+local socket = require "cosock.socket"
+local ssl = require "cosock.ssl"
+
+local jbl_api = {}
+jbl_api.__index = jbl_api
+
+local CREDENTIAL_TIME_OUT_SECONDS = 30
+
+local SSL_CONFIG = {
+    mode = "client",
+    protocol = "any",
+    verify = "peer",
+    options = "all",
+    cafile="./selfSignedRoot.crt"
+}
+
+local ADDITIONAL_HEADERS = {
+    ["Accept"] = "application/json",
+    ["Content-Type"] = "application/json",
+}
+
+function jbl_api.socket_builder(host, port, wrap_ssl)
+    local sock, err = socket.tcp()
+
+    if err ~= nil or (not sock) then
+      return nil, (err or "unknown error creating TCP socket")
+    end
+
+    sock:setoption("keepalive", true)
+    _, err = sock:connect(host, port)
+
+    if wrap_ssl then
+      sock, err =
+        ssl.wrap(sock, SSL_CONFIG)
+      if sock ~= nil then
+        _, err = sock:dohandshake()
+      elseif err ~= nil then
+        log.error("Error setting up TLS: " .. err)
+      end
+    end
+
+    if err ~= nil then sock = nil end
+
+    return sock, err
+end
+
+local function get_base_url(device_ip)
+    return "https://" .. device_ip .. ":4443"
+end
+
+local function process_rest_response(response, err, partial)
+    if err ~= nil then
+        return response, err, nil
+    elseif response ~= nil then
+        return json.decode(response:get_body()), nil, response.status
```

Comment:
This was heavily refactored on our branch because `json.decode` can throw a Lua `error` which will crash the driver. I would suggest doing the same here.

Reply:
change to        

> local status, decoded_json = pcall(json.decode, response:get_body())